src/shims/os_str.rs

```diff
@@ -177,72 +177,91 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         'mir: 'a,
     {
         let this = self.eval_context_ref();
-        let os_str = this.read_os_str_from_c_str(scalar)?;
+        let os_str: &'a OsStr = this.read_os_str_from_c_str(scalar)?;
```

Comment:
Is the type annotation here really needed? `read_os_str_from_c_str` has a fixed return type, so it should be unnecessary.